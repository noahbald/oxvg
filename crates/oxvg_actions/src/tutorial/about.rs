use rand::{seq::SliceRandom, thread_rng};

pub struct About<'a> {
    pub version: &'a str,
    pub authors: Vec<&'a str>,
    pub license: &'a str,
}

pub fn about<'a>() -> About<'a> {
    let mut about = About::default();
    about.authors.shuffle(&mut thread_rng());
    about
}

impl Default for About<'_> {
    fn default() -> Self {
        Self {
            version: "v0.1.0",
            authors: vec!["Noah <noahwbaldwin@gmail.com>"],
            license: "Most oxvg source code is available under the MIT License.\n\nNotable Exceptions are\n- actions based on Inkscape are mostly licensed under the GNU License"
        }
    }
}

#[test]
fn test_about() {
    let about = about();
    insta::assert_snapshot!(about.version);
    insta::assert_snapshot!(about.license);
}
