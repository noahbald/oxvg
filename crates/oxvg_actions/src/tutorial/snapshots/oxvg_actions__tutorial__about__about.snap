---
source: crates/oxvg_actions/src/tutorial/about.rs
expression: about.version
---
v0.1.0
